docs: use flowing verse on debug page explaining DNS server failures (#45141)
home-assistant/home-assistant.github.io: source/more-info/unsupported/dns_server.markdown

## The issue

> Home Assistant requires a working DNS server to function. Without one it may be unable to provide functionality like checking and executing updates, showing documentation, reach external services required by add-ons and integrations, etc.

Author: @Copilot Did this issue arise as a result of this PR? If not, I will most likely open another PR for this change.

> Home Assistant requires a working DNS server to function. Without one it may be unable to provide functionality like checking and executing updates, showing documentation, reach external services required by add-ons and integrations, etc.

Author: @Copilot Don't you think that that would belong in a different PR?

> Home Assistant requires a working DNS server to function. Without one it may be unable to provide functionality like checking and executing updates, showing documentation, reach external services required by add-ons and integrations, etc.

Reviewer: ```suggestion
Home Assistant requires a working DNS server to function. Without one, it may be unable to provide functionality like checking and executing updates, showing documentation, reaching external services required by add-ons and integrations, and other tasks that depend on internet access.
```

### `dns_server_ipv6_error`

> If you see this that means the application you are using for DNS is not handling A and AAAA requests correctly. You can test this by executing the following commands:

Author: @Copilot Do you think that that would belong in a different PR?

> If you see this that means the application you are using for DNS is not handling A and AAAA requests correctly. You can test this by executing the following commands:

Reviewer: ```suggestion
If you see this, that means the application you are using for DNS is not handling A and AAAA requests correctly. You can test this by executing the following commands:
```

```sh
server="<IP address of DNS server>"
dig "@$server" _checkdns.home-assistant.io +noall +comments +answer A
dig "@$server" _checkdns.home-assistant.io +noall +comments +answer AAAA
```

A DNS server handling A and AAAA requests correctly will respond with status `NOERROR` for both of those queries even though there are no answers for the AAAA request. A DNS server mishandling this request will only return a `NOERROR` response for the first one and will return `NXDOMAIN`, `REFUSED`, `SERVFAIL` or some other error status for the second.

> It is important to use a DNS server that handles this situation correctly since Home Assistant uses alpine for many of its containers. Alpine follows the DNS spec and will treat the entire domain as if it can't be resolved if it receives an error status for either query. Home Assistant will run into many unexpected issues in this situation, particularly around updates and installing software.

Author: @Copilot That will most likely be in another PR.

> It is important to use a DNS server that handles this situation correctly since Home Assistant uses alpine for many of its containers. Alpine follows the DNS spec and will treat the entire domain as if it can't be resolved if it receives an error status for either query. Home Assistant will run into many unexpected issues in this situation, particularly around updates and installing software.

Reviewer: ```suggestion
It is important to use a DNS server that handles this situation correctly since Home Assistant uses Alpine Linux for many of its containers. Alpine follows the DNS spec and will treat the entire domain as if it can't be resolved if it receives an error status for either query. Home Assistant will run into many unexpected issues in this situation, particularly around updates and installing software.
```
